packages/grafana-ui/src/components/uPlot/MicroPlot.tsx

```diff
@@ -58,18 +55,14 @@ export class MicroPlot extends PureComponent<Props, State> {
   }
 
   renderLoop = () => {
-    requestAnimationFrame(this.renderLoop);
```

Comment:
@ryantxu why even have a render loop? don't you guys have debounced/throttled hooks for when new data arrives? what's the purpose of wasting cycles doing 20Hz noop redraws? if there's 50k datapoints just sitting there, this won't be nearly as cheap as simply doing nothing.

Reply:
it depends :)   with high frequency consistent updates you are right.  This actually solves keeping the graph moving left when the data is *not* updating.  consider multiple graphs on the same dashboard where they each update at different rates.

I am updating the logic so this is:
1. optional
2. only happens when the selected time range is relative to `now`
3. will set the refresh rate close to the possible pixel updates.   IE:  now-6h in 1000px does not need to update more than every 20s

Reply:
nice 👍

Reply:
> will set the refresh rate close to the possible pixel updates. IE: now-6h in 1000px does not need to update more than every 20s

there might actually be a bit more nuance to this. even if the graph does not move, the per-pixel high/low accumulation uPlot does can still cause the visual graph changes if your polling period < 20s. so i guess it would need to be some combo of *either* new incoming data & setData() *or* 20s have elapsed & setScale() for shifting the displayed range.

Reply:
for sure -- my main point is not trying to render fast unless it is really necessary